### ISSUE TITLE
[npm/yarn] InvalidPackageName

### ISSUE THREAD

**Issue by Sergio-IME** [dependabot/dependabot-core#5246]
Hi
(sorry if this has to go to support instead of here, not quite sure)
it seems that [this particular npm package](https://www.npmjs.com/package/@_sh/strapi-plugin-ckeditor) is causing trouble in Insights > Dep graph > Dependabot tab, but maybe it has other ramifications

**Package ecosystem**
npm

**Images of the diff or a link to the PR, issue, or logs**
```
updater | ERROR <job_XXXXXXXXX> Error processing @_sh/strapi-plugin-ckeditor (Dependabot::NpmAndYarn::PackageName::InvalidPackageName)
updater | ERROR <job_XXXXXXXXX> Dependabot::NpmAndYarn::PackageName::InvalidPackageName
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-npm_and_yarn-0.191.0/lib/dependabot/npm_and_yarn/package_name.rb:26:in `initialize'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-npm_and_yarn-0.191.0/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb:234:in `new'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-npm_and_yarn-0.191.0/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb:234:in `types_package'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-npm_and_yarn-0.191.0/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb:251:in `types_update_available?'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-npm_and_yarn-0.191.0/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb:78:in `latest_resolvable_version'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-npm_and_yarn-0.191.0/lib/dependabot/npm_and_yarn/update_checker.rb:32:in `latest_resolvable_version'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-common-0.191.0/lib/dependabot/update_checkers/base.rb:74:in `preferred_resolvable_version'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-common-0.191.0/lib/dependabot/update_checkers/base.rb:257:in `preferred_version_resolvable_with_unlock?'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-common-0.191.0/lib/dependabot/update_checkers/base.rb:249:in `numeric_version_can_update?'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-common-0.191.0/lib/dependabot/update_checkers/base.rb:199:in `version_can_update?'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-common-0.191.0/lib/dependabot/update_checkers/base.rb:44:in `can_update?'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:454:in `requirements_to_unlock'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:232:in `check_and_create_pull_request'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:82:in `check_and_create_pr_with_error_handling'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:56:in `block in run'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:56:in `each'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:56:in `run'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/lib/dependabot/update_files_job.rb:17:in `perform_job'
updater | ERROR <job_XXXXXXXXX> /home/dependabot/dependabot-updater/lib/dependabot/base_job.rb:35:in `run'
updater | ERROR <job_XXXXXXXXX> bin/update_files.rb:22:in `<main>'
```

**Comment by mattt** [comment]
Thanks so much for reporting this issue, @Sergio-IME.

According to the [`npm-user-validate`](https://github.com/npm/npm-user-validate/) package, the requirement is that usernames contain only "URL-safe" characters (according to `encodeURIComponent`) and not have a leading dot (`.`):

https://github.com/npm/npm-user-validate/blob/dec282efeea3bb711a43ed5a7fdd509934f17410/npm-user-validate.js#L28-L30

However, in `Dependabot::NpmAndYarn::PackageName`, we allow underscores, but not in the first character.

https://github.com/dependabot/dependabot-core/blob/1bc65b5e1020c8eed70acfd01e172159fed6ea05/npm_and_yarn/lib/dependabot/npm_and_yarn/package_name.rb#L9

I'll open a PR with a fix soon.

**Comment by Sergio-IME** [comment]
Hello
`0.193.0` is not happy either
```
updater | ERROR <job_XXXXXXXX> Error processing @_sh/strapi-plugin-ckeditor (Dependabot::NpmAndYarn::PackageName::InvalidPackageName)
updater | ERROR <job_XXXXXXXX> Dependabot::NpmAndYarn::PackageName::InvalidPackageName
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-npm_and_yarn-0.193.0/lib/dependabot/npm_and_yarn/package_name.rb:35:in `initialize'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-npm_and_yarn-0.193.0/lib/dependabot/npm_and_yarn/package_name.rb:80:in `new'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-npm_and_yarn-0.193.0/lib/dependabot/npm_and_yarn/package_name.rb:80:in `types_package_name'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-npm_and_yarn-0.193.0/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb:234:in `types_package'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-npm_and_yarn-0.193.0/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb:251:in `types_update_available?'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-npm_and_yarn-0.193.0/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb:78:in `latest_resolvable_version'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-npm_and_yarn-0.193.0/lib/dependabot/npm_and_yarn/update_checker.rb:32:in `latest_resolvable_version'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-common-0.193.0/lib/dependabot/update_checkers/base.rb:74:in `preferred_resolvable_version'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-common-0.193.0/lib/dependabot/update_checkers/base.rb:257:in `preferred_version_resolvable_with_unlock?'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-common-0.193.0/lib/dependabot/update_checkers/base.rb:249:in `numeric_version_can_update?'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-common-0.193.0/lib/dependabot/update_checkers/base.rb:199:in `version_can_update?'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/vendor/ruby/2.7.0/gems/dependabot-common-0.193.0/lib/dependabot/update_checkers/base.rb:44:in `can_update?'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:454:in `requirements_to_unlock'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:232:in `check_and_create_pull_request'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:82:in `check_and_create_pr_with_error_handling'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:56:in `block in run'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:56:in `each'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:56:in `run'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/lib/dependabot/update_files_job.rb:17:in `perform_job'
updater | ERROR <job_XXXXXXXX> /home/dependabot/dependabot-updater/lib/dependabot/base_job.rb:35:in `run'
updater | ERROR <job_XXXXXXXX> bin/update_files.rb:22:in `<main>'
```

**Comment by abdulapopoola** [comment]
Fixed in #9569
